# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/cache/FileUtils.h
# Commit: 89df3c99bf3b
# Full Hash: 89df3c99bf3bb8ba1918b2685ac8fa4df6ff9e98
# Author: Tom Tung <shes050117@gmail.com>
# Date: 2019-09-28 09:46:05
# Regressor Bug: 1581067
# File Overlap Count: 1
# Description:
#   Bug 1581067 - P2 - Extract similar to reuse the code; r=asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D46780
# ==============================================================================

diff -r fbfdd8ee1b42 -r 89df3c99bf3b dom/cache/FileUtils.h
--- a/dom/cache/FileUtils.h	Fri Sep 27 16:54:22 2019 +0000
+++ b/dom/cache/FileUtils.h	Fri Sep 27 16:54:45 2019 +0000
@@ -58,6 +58,11 @@
 nsresult BodyDeleteOrphanedFiles(const QuotaInfo& aQuotaInfo, nsIFile* aBaseDir,
                                  nsTArray<nsID>& aKnownBodyIdList);
 
+template <typename Func>
+nsresult BodyTraverseFiles(const QuotaInfo& aQuotaInfo, nsIFile* aBodyDir,
+                           const Func& aHandleFileFunc,
+                           const bool aTrackQuota = true);
+
 nsresult CreateMarkerFile(const QuotaInfo& aQuotaInfo);
 
 nsresult DeleteMarkerFile(const QuotaInfo& aQuotaInfo);